src/Wellcome.Dds/Wellcome.Dds.AssetDomainRepositories/Ingest/CloudServicesIngestRegistry.cs

```diff
@@ -0,0 +1,179 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Threading.Tasks;
+using Wellcome.Dds.AssetDomain.Dlcs.Ingest;
+using Wellcome.Dds.AssetDomain.Mets;
+
+namespace Wellcome.Dds.AssetDomainRepositories.Ingest
+{
+
+    public class CloudServicesIngestRegistry : IIngestJobRegistry
+    {
+        private readonly IMetsRepository _metsRepository;
+        private readonly DdsInstrumentationContext ddsInstrumentationContext;
+
+        public CloudServicesIngestRegistry(
+            IMetsRepository metsRepository,
+            DdsInstrumentationContext ddsInstrumentationContext)
+        {
+            _metsRepository = metsRepository;
+            this.ddsInstrumentationContext = ddsInstrumentationContext;
+        }
+
+        public IEnumerable<DlcsIngestJob> GetJobs(int number)
+        {
+            return ddsInstrumentationContext.DlcsIngestJobs
+                .OrderByDescending(j => j.Created)
+                .Take(number)
+                .Include(j => j.DlcsBatches)
+                .ToArray(); // The operation cannot be completed because the DbContext has been disposed. Want to defer this
+        }
+
+        public DlcsIngestJob GetJob(int id)
+        {
+            return ddsInstrumentationContext.DlcsIngestJobs
+                .Include(j => j.DlcsBatches)
+                .FirstOrDefault(j => j.Id == id);
```

Comment:
Would `.SingleOrDefault(j => j.Id == id);` be safer here? Assuming there shouldn't be >1 with same Id

Reply:
yes, have changed